lexers/LexCPP.cxx

```diff
@@ -365,8 +364,8 @@ struct OptionsCPP {
 		foldComment = false;
 		foldCommentMultiline = true;
 		foldCommentExplicit = true;
-		foldExplicitStart = "";
-		foldExplicitEnd = "";
+		foldExplicitStart.clear();
```

Comment:
I think it's redundant to clear these `std::string` fields in the constructor. The constructor can even be removed by applying [`modernize-use-default-member-init`](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-default-member-init.html).

Reply:
The point of Options\* initialisers is to make it easy for a maintainer to see what the options are and what their default values are. A default can be easily changed just by editing the value assigned. They are constructed once at lexer creation so there is no efficiency concern.

With C++11, in-class member initializers (`modernize-use-default-member-init` ) are available, so this may be clearer and less verbose as:
~~~c++
struct OptionsCPP {
	bool stylingWithinPreprocessor = false;
	bool identifiersAllowDollars = true;
	//...
~~~
`modernize-use-default-member-init` prefers braced initializers which are ugly for simple cases which alost all Options structs will be.

It is easy to make a mistake with modernization and there is little gain.

Reply:
@zufuliu, if you stick to clang-tidy, then you said correctly.

@nyamatongwe, should I change to `modernize-use-default-member-init` or not?

Reply:
> should I change to modernize-use-default-member-init or not?

To me there is insufficient net benefit and there aren't test cases to check that the changes don't cause a behavioural regression.